flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/windowing/ContinuousProcessingTimeTriggerTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.operators.windowing;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.streaming.api.windowing.time.Time;
+import org.apache.flink.streaming.api.windowing.triggers.ContinuousProcessingTimeTrigger;
+import org.apache.flink.streaming.api.windowing.triggers.TriggerResult;
+import org.apache.flink.streaming.api.windowing.windows.TimeWindow;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+
+import org.apache.flink.shaded.guava18.com.google.common.collect.Lists;
+
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+/** Tests for {@link ContinuousProcessingTimeTrigger}. */
+public class ContinuousProcessingTimeTriggerTest {
+    @Test
+    public void testMergingWindows() throws Exception {
```

Comment:
Could we use `KeyedOneInputStreamOperatorTestHarness` instead of modifying the `TriggerTestHarness`?
You should be able to advance processing time via: `testHarness.getProcessingTimeService().setCurrentTime()`.

TBH, I'd feel safer if we do not need to modify the `TriggerTestHarness`.

Reply:
I will give it a try. How about keep both ? This way we can achieve two in one pr:
* Align `TriggerTestHarness` with real operator test harness. (eg. ignore timer for merged out window)
* Test `ContinuousProcessingTimerTrigger` in both `TriggerTestHarness` and operator test harness.

Reply:
I'd rather not touch the `TriggerTestHarness`. I don't understand the class and would have to spend quite some time reading it. Moreover I don't see we change it often in the future. I'd feel much safer without modifying the class.

Reply:
I understand the situation. I will keep it untouched.

Reply:
A read operator provides more confident than sole trigger harness.